src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -117,31 +117,37 @@ public static SqlGuid Parse(string s)
                 return new SqlGuid(s);
         }
 
-
         // Comparison operators
         private static EComparison Compare(SqlGuid x, SqlGuid y)
         {
             // Comparison orders.
             ReadOnlySpan<byte> rgiGuidOrder = new byte[16] { 10, 11, 12, 13, 14, 15, 8, 9, 6, 7, 4, 5, 0, 1, 2, 3 };
 
+            Debug.Assert(!x.IsNull);
+            Debug.Assert(!y.IsNull);
+
             // Swap to the correct order to be compared
-            ReadOnlySpan<byte> xBytes = x.m_value;
-            ReadOnlySpan<byte> yBytes = y.m_value;
+            Span<byte> xBytes = stackalloc byte[sizeof(Guid)];
+            bool xWrote = x._value.GetValueOrDefault().TryWriteBytes(xBytes);
+            Debug.Assert(xWrote);
+
+            Span<byte> yBytes = stackalloc byte[sizeof(Guid)];
+            bool yWrote = y._value.GetValueOrDefault().TryWriteBytes(yBytes);
+            Debug.Assert(yWrote);
+
             for (int i = 0; i < SizeOfGuid; i++)
             {
                 byte b1 = xBytes[rgiGuidOrder[i]];
                 byte b2 = yBytes[rgiGuidOrder[i]];
                 if (b1 != b2)
                 {
-                    return (b1 < b2) ? EComparison.LT : EComparison.GT;
+                    return b1 < b2 ? EComparison.LT : EComparison.GT;
```

Comment:
Keep the parens on this and later methods in this file to help reduce the size of the code change.